src/workerd/io/worker.c++

```diff
@@ -3356,20 +3355,41 @@ kj::Promise<void> Worker::Isolate::SubrequestClient::request(
     }
     headersToCDP(headers, response.initHeaders());
 
-    i.sendNotification(event);
-
     auto encoding = api::StreamEncoding::IDENTITY;
     KJ_IF_MAYBE(encodingStr, headers.get(contentEncodingHeaderId)) {
       if (*encodingStr == "gzip") {
         encoding = api::StreamEncoding::GZIP;
       }
     }
 
-    return kj::heap<ResponseStreamWrapper>(kj::atomicAddRef(*constIsolate),
-                                           kj::mv(requestId),
-                                           kj::mv(responseBody),
-                                           encoding,
-                                           *requestMetrics);
+    // Defer to a later turn of the event loop so that it's safe to take a lock.
+    return kj::newPromisedStream(kj::evalLater(
```

Comment:
Is there some way we could determine if we had the inspector session without taking the isolate lock? It would be nice if we could avoid the promised stream if we know there is no inspector and I notice the comment about `signalRequest()`. That said, the captures look right to me.

Reply:
As mentioned in the comments, in practice we won't reach this code in the first place if there's no inspector session.

I do think this code could use a cleanup but not volunteering for that at the moment.